src/lib.rs

```diff
@@ -1249,15 +1272,17 @@ impl<T: Counter> Histogram<T> {
     /// covered by previous buckets), and the +1 being used for the lower half of the 0'th bucket.
     /// Or, equivalently, we need 1 more bucket to capture the max value if we consider the
     /// sub-bucket length to be halved.
-    fn num_bins(&self, number_of_buckets: usize) -> usize {
-        (number_of_buckets + 1) * (self.sub_bucket_count / 2)
+    fn num_bins(&self, number_of_buckets: u8) -> u32 {
+        // TODO use sub_bucket_half_count
+        (number_of_buckets as u32 + 1) * (self.sub_bucket_count / 2)
```

Comment:
There might be a subtle issue with this being a "cold" field, so I'll see if this gets accepted as an upstream PR before I do it here. (Also, would be interesting to see if we can get a win by controlling field order once we get to concurrent histograms.)